src/v2/Apps/_Preferences2/PreferencesApp.tsx

```diff
@@ -1,9 +1,182 @@
-import { Text } from "@artsy/palette"
+import {
+  Box,
+  Button,
+  Checkbox,
+  Column,
+  Flex,
+  Separator,
+  Spacer,
+  Text,
+} from "@artsy/palette"
+import { useState } from "react"
 
 export const PreferencesApp: React.FC = () => {
+  const [uncheckedAll, setUncheckedAll] = useState(false)
```

Comment:
I suspect we'll wind up using Formik for managing the form's state.

Reply:
How so?

Reply:
Something like:

```tsx
<Formik
  initialValues={{
    first: false,
    second: false,
    third: false,
  }}
  onSubmit={values => {
    console.log(values)
  }}
>
  {({ values, setFieldValue }) => {
    return (
      <>
        <Checkbox
          selected={Object.values(values).every(Boolean)}
          onSelect={value => {
            ;["first", "second", "third"].forEach(field => {
              setFieldValue(field, value)
            })
          }}
        >
          Select all
        </Checkbox>

        <Checkbox
          selected={values.first}
          onSelect={value => {
            setFieldValue("first", value)
          }}
        >
          First
        </Checkbox>

        <Checkbox
          selected={values.second}
          onSelect={value => {
            setFieldValue("second", value)
          }}
        >
          Second
        </Checkbox>

        <Checkbox
          selected={values.third}
          onSelect={value => {
            setFieldValue("third", value)
          }}
        >
          Third
        </Checkbox>

        <Checkbox
          selected={Object.values(values).every(v => !v)}
          onSelect={value => {
            ;["first", "second", "third"].forEach(field => {
              setFieldValue(field, !value)
            })
          }}
        >
          Unselect all
        </Checkbox>
      </>
    )
  }}
</Formik>
```

Reply:
Sort of an interesting question is what do you do about UNCHECKING the 'all' checkboxes?:

https://user-images.githubusercontent.com/112297/150876189-1f454399-cf6b-476b-ab39-0dcd3374dcd9.mov

Reply:
Cleared it with Jenna, we want to have all the other checkboxes selected again. Moved all this work over to this PR: https://github.com/artsy/force/pull/9253